packages/now-routing-utils/src/index.ts

```diff
@@ -100,79 +92,85 @@ export function normalizeRoutes(inputRoutes: Route[] | null): NormalizedRoutes {
       const handleValue = handling[handling.length - 1];
       if (handleValue === 'hit') {
         if (route.dest) {
-          errors.push({
-            message: `You cannot assign "dest" after "handle: hit"`,
-            src: route.src,
-          });
+          errors.push(
+            `Route at index ${i} cannot define "dest" after "handle: hit".`
+          );
         }
         if (route.status) {
-          errors.push({
-            message: `You cannot assign "status" after "handle: hit"`,
-            src: route.src,
-          });
+          errors.push(
+            `Route at index ${i} cannot define "status" after "handle: hit".`
+          );
         }
         if (!route.continue) {
-          errors.push({
-            message: `You must assign "continue: true" after "handle: hit"`,
-            src: route.src,
-          });
+          errors.push(
+            `Route at index ${i} must define "continue: true" after "handle: hit".`
+          );
         }
       } else if (handleValue === 'miss') {
         if (route.dest && !route.check) {
-          errors.push({
-            message: `You must assign "check: true" after "handle: miss"`,
-            src: route.src,
-          });
+          errors.push(
+            `Route at index ${i} must define "check: true" after "handle: miss".`
+          );
         } else if (!route.dest && !route.continue) {
-          errors.push({
-            message: `You must assign "continue: true" after "handle: miss"`,
-            src: route.src,
-          });
+          errors.push(
+            `Route at index ${i} must define "continue: true" after "handle: miss".`
+          );
         }
       }
     } else {
-      errors.push({
-        message: 'A route must set either handle or src',
-      });
+      errors.push(
+        `Route at index ${i} must define either "handle" or "src" property.`
+      );
     }
-  }
+  });
 
-  const error = createNowError(
-    'invalid_routes',
-    'One or more invalid routes were found',
-    errors
-  );
+  const error =
+    errors.length > 0
+      ? createError(
+          'invalid_route',
+          errors[0],
```

Comment:
Why do we only return the first error here?

Reply:
We don't have a good support for multi-line errors right now.

Today, this error could be displayed in 4+ places:

- cli (works with multiline)
- git integration (does not - truncated to first line)
- build logs (kinda of works, not highlighted properly)
- summary at the top left (does not work - newlines are not preserved)

Reply:
The other reason is that each error could have a separate link for the "Learn More"

Reply:
Could we provide the full errors to the environments that do have multi-line support and then provide the first error to the environments that don't?

Reply:
No because the builds logs and the summary at the top share the same `error.message` value

![image](https://user-images.githubusercontent.com/229881/83278110-10177200-a1a1-11ea-99f9-e49f4eeb6768.png)

It is unlikely that a user will have multiple routes with errors since the one in the screenshot has been combined into a single error anyway.

Reply:
I see, if a user does have multiple routes with errors though it seems like it would be a pretty painful experience to fix one re-deploy and have the next one show so and continue for however many incorrect routes they do have. If we could avoid this I think it would be ideal

Reply:
The user doesn't need to deploy, it will be caught with `vercel dev` too